openedx/features/qverse_features/registration/signals.py

```diff
@@ -131,17 +129,23 @@ def _create_or_update_qverse_user_profile(edx_user, user_info):
         edx_user (User): An edx user instance whose qverse profile is going to be created/updated
         user_info (dict): User info whose profile is going to be created/updated
     """
-    department = Department.objects.get(number=user_info.get('departmentid'))
+    department = None
+    try:
+        department = Department.objects.get(number=user_info.get('departmentid'))
+    except Department.DoesNotExist:
+        LOGGER.exception('The Department with id {} does not exist.'.format(user_info.get('departmentid')))
+        user_info.update({'status': 'Failed'})
```

Comment:
no need to update you can simply reassigned value `row['status'] = 'Failed'`

Reply:
done.